zeton/api/user.py

```diff
@@ -0,0 +1,37 @@
+from flask import request, url_for, g
+from werkzeug.security import generate_password_hash
+from werkzeug.utils import redirect
+
+from zeton.api import bp
+from zeton.data_access import users
+from zeton.data_access.bans import insert_all_default_bans
+from zeton.data_access.users import add_new_user, get_child_id, associate_child_with_caregiver
+
+
+@bp.route("/user", methods=['POST'])
+def register():
+    users.load_logged_in_user_data()
+    username = request.form['username']
+    password = request.form['password']
+    password_hash = generate_password_hash(password)
+    role = request.form.get('role') or 'caregiver'
+    firstname = request.form.get('name') or '-'
+
+    data = (username, password_hash, role, firstname)
+    try:
+        add_new_user(data)
+    except Exception as ex:
```

Comment:
Łapanie czegokolwiek i dawanie 400 - bad request, to nie jest dobra praktyka. 400 mówi, że samo zapytanie jest źle utworzone.
Zamiast `Exception` powinien być konkretny wyjątek.

Reply:
Jasna sprawa, mój błąd, to pozostałość po debugowaniu - chciałem sprawdzić jakie wyjątki w ogóle można złapać. Będzie poprawione.